tests/test_repository.py

```
@@ -157,3 +172,384 @@ def test_clone_real_failure_handling(self, mock_repo: MagicMock) -> None:
+        manager = RepositoryManager()
+        target = tmp_path / "my-repo"
+        manager.clone_into("https://github.com/org/repo", target)
+        assert target not in manager.temp_dirs
+
+    def test_clone_into_rejects_file_url(self) -> None:
+        """Test that clone_into rejects file:// URLs."""
+        from pathlib import Path
+
+        manager = RepositoryManager()
+        with pytest.raises(ValueError, match="Only https:// and git://"):
+            manager.clone_into("file:///etc/passwd", Path("/tmp/test"))
```

Ruff (0.15.7)

[error] Probable insecure usage of temporary file or directory: "/tmp/test"

(S108)